mllib/src/main/scala/org/apache/spark/ml/classification/LinearSVC.scala

```diff
@@ -463,6 +458,8 @@ private class LinearSVCAggregator(
    */
   def add(instance: Instance): this.type = {
     instance match { case Instance(label, weight, features) =>
+      require(numFeatures == features.size, s"Dimensions mismatch when adding new instance." +
```

Comment:
So we are moving the dimension check here yes? Seems better than at [L446](https://github.com/apache/spark/pull/17076/files#diff-0d3dbd8bcb54f614839418d64680593aL446) in fact